### PR TITLE
Fix a false positive and incorrect autocorrect for `Capybara/SpecificActions`, `Capybara/SpecificFinders` and `Capybara/SpecificMatcher`

### PR DESCRIPTION
**Pull request by ydah**
Fix: https://github.com/rubocop/rubocop-capybara/issues/4

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `main` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [-] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: spec/rubocop/cop/capybara/mixin/css_selector_spec.rb [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
```
Comment (ydah) [inline spec/rubocop/cop/capybara/mixin/css_selector_spec.rb:1]: [memo] Add mixin test.
https://github.com/rubocop/rubocop-rspec/pull/1481#pullrequestreview-1215220830

File: lib/rubocop/cop/capybara/mixin/css_selector.rb [hunk 2]
```diff
@@ -2,141 +2,107 @@
...
-      def common_attributes?(selector)
-        attributes(selector).keys.difference(COMMON_OPTIONS).none?
-      end
+        # @param selector [String]
+        # @return [Array<String>]
+        # @example
+        #   attributes('a[foo-bar_baz]') # => {"foo-bar_baz=>true}
+        #   attributes('button[foo][bar]') # => {"foo"=>true, "bar"=>true}
+        #   attributes('table[foo=bar]') # => {"foo"=>"'bar'"}
+        def attributes(selector)
```
Comment (ydah) [inline lib/rubocop/cop/capybara/mixin/css_selector.rb:57]: [memo] It got a little complicated, so I changed the logic while adding comments.
https://github.com/rubocop/rubocop-rspec/pull/1481#discussion_r1046986310

File: lib/rubocop/cop/capybara/specific_matcher.rb [hunk 3]
```diff
@@ -39,15 +39,14 @@ class SpecificMatcher < ::RuboCop::Cop::Base
 
         # @!method first_argument(node)
         def_node_matcher :first_argument, <<-PATTERN
-          (send nil? _ (str $_) ... )
+            (send nil? _ (str $_) ... )
```
Comment (pirj) [inline lib/rubocop/cop/capybara/specific_matcher.rb:42]:
```suggestion
          (send nil? _ (str $_) ... )
```

**Review by pirj — APPROVED** [review]
I see no flaws.
Just glanced over - everything looks good.
Please go ahead with merging as you see fit.

I see some major changes in specs e.g. `[disabled]` -> `[disabled=true]`, but I trust your judgement that this is correct.

Thank you for such an effort! :raised_hands:

**Comment by pirj** [comment]
Merge at will, @ydah !

**Pull request merged**